test/e2e/apps/disruption.go

```diff
@@ -335,3 +361,24 @@ func createReplicaSetOrDie(cs kubernetes.Interface, ns string, size int32, exclu
 	_, err := cs.AppsV1().ReplicaSets(ns).Create(rs)
 	framework.ExpectNoError(err, "Creating replica set %q in namespace %q", rs.Name, ns)
 }
+
+func locateARunningPod(cs kubernetes.Interface, ns string) (pod v1.Pod, err error) {
+	By("locating a running pod")
+	err = wait.PollImmediate(framework.Poll, schedulingTimeout, func() (bool, error) {
+		podList, err := cs.CoreV1().Pods(ns).List(metav1.ListOptions{})
+		if err != nil {
+			return false, err
+		}
+
+		for i := range podList.Items {
+			if podList.Items[i].Status.Phase == v1.PodRunning {
+				pod = podList.Items[i]
+				return true, nil
+			}
+		}
+
+		return false, nil
+	})
+	framework.ExpectNoError(err, "locating a running pod in namespace %q", ns)
```

Comment:
Since this returns the error, let's leave it up to the caller to check it. Otherwise we're handling it twice.

Reply:
removed